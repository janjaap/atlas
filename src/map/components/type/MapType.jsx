import React from 'react';
import PropTypes from 'prop-types';

import { SelectButton } from '../../../shared/components/select-button';

import './_map-type.scss';

function hasLayer(activeBaseLayer, baseLayers) {
  return baseLayers && baseLayers.length > 0 &&
    baseLayers.some((layer) => layer.value === activeBaseLayer);
}

const MapType = ({ activeBaseLayer, baseLayers, onBaseLayerToggle }) => (
  <section className="map-type">
    <h3 className="map-type__title">Achtergrond</h3>

    <SelectButton
      className="map-type__select map-type__select--topography"
      handleChange={onBaseLayerToggle}
      icon={<span className="icon icon--topography" />}
      isDisabled={!hasLayer(activeBaseLayer, baseLayers.topography)}
      name="topography"
      options={baseLayers.topography}
      value={hasLayer(activeBaseLayer, baseLayers.topography) ? activeBaseLayer : null}
    />

    <SelectButton
      className="map-type__select map-type__select--aerial"
      handleChange={onBaseLayerToggle}
      icon={<span className="icon icon--aerial" />}
      isDisabled={!hasLayer(activeBaseLayer, baseLayers.aerial)}
      name="aerial"
      options={baseLayers.aerial}
      value={hasLayer(activeBaseLayer, baseLayers.aerial) ? activeBaseLayer : null}
    />
  </section>
);

MapType.propTypes = {
  activeBaseLayer: PropTypes.string.isRequired,
  baseLayers: PropTypes.object, // eslint-disable-line
  onBaseLayerToggle: PropTypes.func.isRequired
};

export default MapType;
