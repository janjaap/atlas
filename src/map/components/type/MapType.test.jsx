import React from 'react';
import { shallow } from 'enzyme';

import MapType from './MapType';

describe('MapType rendering', () => {
  const baseLayers = {
    topography: [
      {
        value: 'topografie',
        label: 'Topografie',
        selected: true,
        urlTemplate: 'https://{s}.data.amsterdam.nl/topo_rd/{z}/{x}/{y}.png'
      }, {
        value: 'topo_rd_light',
        label: 'Topografie licht',
        urlTemplate: 'https://{s}.data.amsterdam.nl/topo_rd_light/{z}/{x}/{y}.png'
      }, {
        value: 'topo_rd_zw',
        label: 'Topografie grijs',
        urlTemplate: 'https://{s}.data.amsterdam.nl/topo_rd_zw/{z}/{x}/{y}.png'
      }
    ],
    aerial: [
      {
        value: 'lf2018',
        label: 'Luchtfoto 2018',
        selected: true,
        urlTemplate: 'https://{s}.data.amsterdam.nl/lufo2018_RD/{z}/{x}/{y}.jpeg'
      }, {
        value: 'ir2018',
        label: 'Infrarood 2018',
        urlTemplate: 'https://{s}.data.amsterdam.nl/infrarood2018_RD/{z}/{x}/{y}.jpeg'
      }, {
        value: 'lf2017',
        label: 'Luchtfoto 2017',
        urlTemplate: 'https://{s}.data.amsterdam.nl/lufo2017_RD/{z}/{x}/{y}.jpeg'
      }, {
        value: 'lf2016',
        label: 'Luchtfoto 2016',
        urlTemplate: 'https://{s}.data.amsterdam.nl/lufo2016_RD/{z}/{x}/{y}.jpeg'
      }, {
        value: 'lf2015',
        label: 'Luchtfoto 2015',
        urlTemplate: 'https://{s}.data.amsterdam.nl/lufo2015_RD/{z}/{x}/{y}.jpeg'
      }
    ]
  };

  it('should render select buttons with topography value', () => {
    const toggleHandler = jest.fn();
    const wrapper = shallow(
      <MapType
        activeBaseLayer="topo_rd_light"
        baseLayers={baseLayers}
        onBaseLayerToggle={toggleHandler}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render select buttons with aerial value', () => {
    const toggleHandler = jest.fn();
    const wrapper = shallow(
      <MapType
        activeBaseLayer="lf2015"
        baseLayers={baseLayers}
        onBaseLayerToggle={toggleHandler}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
