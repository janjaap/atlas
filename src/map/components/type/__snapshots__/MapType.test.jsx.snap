// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapType rendering should render select buttons with aerial value 1`] = `
<section
  className="map-type"
>
  <h3
    className="map-type__title"
  >
    Achtergrond
  </h3>
  <SelectButton
    className="map-type__select map-type__select--topography"
    handleChange={[MockFunction]}
    icon={[Function]}
    isDisabled={true}
    isExpanded={false}
    label=""
    name="topography"
    options={
      Array [
        Object {
          "label": "Topografie",
          "selected": true,
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd/{z}/{x}/{y}.png",
          "value": "topografie",
        },
        Object {
          "label": "Topografie licht",
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd_light/{z}/{x}/{y}.png",
          "value": "topo_rd_light",
        },
        Object {
          "label": "Topografie grijs",
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd_zw/{z}/{x}/{y}.png",
          "value": "topo_rd_zw",
        },
      ]
    }
    value={null}
  />
  <SelectButton
    className="map-type__select map-type__select--aerial"
    handleChange={[MockFunction]}
    icon={[Function]}
    isDisabled={false}
    isExpanded={false}
    label=""
    name="aerial"
    options={
      Array [
        Object {
          "label": "Luchtfoto 2017",
          "selected": true,
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2017_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2017",
        },
        Object {
          "label": "Luchtfoto 2016",
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2016_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2016",
        },
        Object {
          "label": "Luchtfoto 2015",
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2015_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2015",
        },
      ]
    }
    value="lf2015"
  />
</section>
`;

exports[`MapType rendering should render select buttons with topography value 1`] = `
<section
  className="map-type"
>
  <h3
    className="map-type__title"
  >
    Achtergrond
  </h3>
  <SelectButton
    className="map-type__select map-type__select--topography"
    handleChange={[MockFunction]}
    icon={[Function]}
    isDisabled={false}
    isExpanded={false}
    label=""
    name="topography"
    options={
      Array [
        Object {
          "label": "Topografie",
          "selected": true,
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd/{z}/{x}/{y}.png",
          "value": "topografie",
        },
        Object {
          "label": "Topografie licht",
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd_light/{z}/{x}/{y}.png",
          "value": "topo_rd_light",
        },
        Object {
          "label": "Topografie grijs",
          "urlTemplate": "https://{s}.data.amsterdam.nl/topo_rd_zw/{z}/{x}/{y}.png",
          "value": "topo_rd_zw",
        },
      ]
    }
    value="topo_rd_light"
  />
  <SelectButton
    className="map-type__select map-type__select--aerial"
    handleChange={[MockFunction]}
    icon={[Function]}
    isDisabled={true}
    isExpanded={false}
    label=""
    name="aerial"
    options={
      Array [
        Object {
          "label": "Luchtfoto 2017",
          "selected": true,
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2017_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2017",
        },
        Object {
          "label": "Luchtfoto 2016",
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2016_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2016",
        },
        Object {
          "label": "Luchtfoto 2015",
          "urlTemplate": "https://{s}.data.amsterdam.nl/lufo2015_RD/{z}/{x}/{y}.jpeg",
          "value": "lf2015",
        },
      ]
    }
    value={null}
  />
</section>
`;
